### ISSUE TITLE
[Question]: Russian censorship

### ISSUE THREAD

**Issue by k0zyrev** [Sommerwiesel/invidious#11]
> This instance is currently under attack by the russian bureau of propaganda and censorship due to hosting videos about the war in ukraine that their totalitarian regime would like to be taken down.
>
> The videos will not be taken down. Freedom of speech will prevail. We stand with Ukraine.

Could you tell a bit more about Roskomnadzor demands? I assume they've sent you an email asking to delete something? Or was it an email to your hosting provider? I'm curious what's all that about, if that was an automatic detection or are they specifically targeting invidious instances. Are you aware of other instance owners receiving such demands?

Obviously, they have no means to shut down your service, but they can easily block your website from within Russia. TBH, invidious was my main plan in case of total youtube blackout. If they target the mirrors, it might be a bad sign.

**Comment by Sommerwiesel** [comment]
Hello there,

Many instances and other private frontends are currently receiving said mails.
And eversince yesterday, my server got hit by several DDoS attacks, all originating from russian ips. I have no proof that it's Roskomnadzor but the timing is too perfect to be a mere coincidence.
As of right now, there are no plans from my side to lockdown the instance from russian visitors.

If Russia decides to blacklist my instance, then there are multiple ways to bypass this:

- use a vpn
- use an encrypted dns server
- use the tor network (various instances from the public list have an onion adress).

TLDR: They can't do shit to block us out completely.
I just wanted to inform visitors of my instance about whats happening because I believe in full transparency.

Cheers

**Comment by k0zyrev** [comment]
Thanks for answering, that's actually valuable info. 

Recently one of the state-owned companies was caught testing a complete youtube mirror solution (possibly based on invidious or something similar), which looked identical to the real thing, but obviously we expect it to have content filters/censorship. In this context them trying to shut down invidious instances is a bit worrisome. So far it indeed looks like automated crawler, so maybe we're not close to youtube ban in Russia yet.

> If Roskomnadzor decides to blacklist my instance, then there are multiple ways to bypass this:

* Also recently there were rather big tests shutting down vpn protocols - wireguard and openvpn. Like blanket bans, doesn't matter if it's business or not, if vpn provider banned or not. It was just not working for a number of users. We already have a shadowsocks solution, but the one of the big problems is spreading the knowledge. BTW, there's also a law forbidding publishing info on how to use vpns (or maybe there's only planning it, I don't quite remember). 
* DNS over HTTPS often doesn't work, they block any connections from blacklisted IPs. [GoodbyeDPI ](https://github.com/ValdikSS/GoodbyeDPI) project might help with that, but I haven't experimented with it.
* if the instance is public, they would have no issues blocking all of them
* YES! This might be the most straightforward way to go. As far as I know, TOR still works even in China, so we should be fine.

Again, thanks for answering, and big thanks for supporting Ukraine and supporting freedom of speech.

**Comment by Sommerwiesel** [comment]
@k0zyrev TOR is now available for my instance (I'm still new to hosting a tor service, unexpected errors might occur): jemgkaq2xibfu37hm2xojsxoi7djtwb25w6krhl63lhn52xfzgeyc2ad.onion

**Comment by k0zyrev** [comment]
Thank you! Very timely indeed: at least from my place, the instance is not available any more. 

I've tried using using the instance through .onion link, and there are some issues. I can't log in: when I fill in my credentials, nothing happens, I get redirected back to the main page, nor I can't save any changed setting.

**Comment by Sommerwiesel** [comment]
@k0zyrev I have created a new thread for the TOR support of my instance [here](https://github.com/Sommerwiesel/invidious/issues/12)

I will add your problems to my todo list.

**Comment by IRainman** [comment]
Thank you so much 🖖🏻 I really like to see civilian activity on the Internet. I'm from Russia, and I'm a programmer and network engineer. My comment here only for the big gratefulness.

**Comment by ifsdk** [comment]
I have all the invidious instances banned since last month. Recently, few months ago working sites were only a couple of the instаnces, then just one.
In most cases the visible to me error is Secure Connection Failed
